src/averageAges.js

```diff
@@ -56,6 +80,21 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  const children = people.filter(person => people.some(mother => mother.name
+    === person.mother));
+  const sons = children.filter(person => checkPersonSex(person, 'm'));
+  let childrenToCount = children;
+
+  if (onlyWithSon) {
+    childrenToCount = sons;
+  }
+
+  const differenceAges = childrenToCount
+    .reduce((summedAges, child) => summedAges
+    + (child.born - people.find(person => person.name
+      === child.mother).born), 0);
```

Comment:
let's make it readable
```suggestion
  const differenceAges = childrenToCount
    .reduce((summedAges, child) => {
    	const mother = ...
    	const ... = child.born - mother.born;
    	
    	return summedAges + ...;
    }, 0);
```

Reply:
done